sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/HDFSBackedStateStoreProvider.scala

```diff
@@ -278,14 +278,15 @@ private[state] class HDFSBackedStateStoreProvider(
 
   /** Initialize the store provider */
   private def initialize(): Unit = {
-    if (!fs.exists(baseDir)) {
-      fs.mkdirs(baseDir)
-    } else {
-      if (!fs.isDirectory(baseDir)) {
+    try {
+      if (!fs.getFileStatus(baseDir).isDirectory()) {
```

Comment:
good point.

Looking at the mkdirs tests {{org.apache.hadoop.fs.contract.AbstractContractMkdirTest}}; the return value indicates whether a directory was created (true) or an existing directory was left alone (false). Other problems (dest is a file, parent path element is a file) are required to raise an IOE or subclass, something that is tested for. All that's needed is to do the mkdirs and not worry about the result at all.